api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentGroupController.java

```diff
@@ -332,6 +361,180 @@ private Payment getPayment(PaymentFeeLink paymentFeeLink, String paymentReferenc
+                .description("Cheque")
+                .build())
+            .paymentStatus(bulkScanPaymentRequestStrategic.getPaymentStatus())
+            .siteId(bulkScanPaymentRequestStrategic.getSiteId())
+            .giroSlipNo(bulkScanPaymentRequestStrategic.getGiroSlipNo())
+            .reportedDateOffline(DateTime.parse(bulkScanPaymentRequestStrategic.getBankedDate()).withZone(DateTimeZone.UTC).toDate())
+            .paymentChannel(bulkScanPaymentRequestStrategic.getPaymentChannel())
+            .documentControlNumber(bulkScanPaymentRequestStrategic.getDocumentControlNumber())
+            .payerName(bulkScanPaymentRequestStrategic.getPayerName())
+            .bankedDate(DateTime.parse(bulkScanPaymentRequestStrategic.getBankedDate()).withZone(DateTimeZone.UTC).toDate())
+            .build();
+
+        PaymentFeeLink paymentFeeLink = paymentGroupService.addNewPaymenttoExistingPaymentGroup(payment, paymentGroupReference);
+
+        Payment newPayment = getPayment(paymentFeeLink, payment.getReference());
+
+        // trigger Apportion based on the launch darkly feature flag
+        boolean apportionFeature = featureToggler.getBooleanValue("apportion-feature",false);
+        LOG.info("ApportionFeature Flag Value in PaymentGroupController  RecordBulkScanPaymentStrategic: {}", apportionFeature);
+        if(apportionFeature) {
+            feePayApportionService.processApportion(newPayment);
+
+            // Update Fee Amount Due as Payment Status received from Bulk Scan Payment as SUCCESS
+            if(newPayment.getPaymentStatus().getName().equalsIgnoreCase("success")) {
+                LOG.info("Update Fee Amount Due as Payment Status received from Bulk Scan Payment as SUCCESS!!!");
+                feePayApportionService.updateFeeAmountDue(newPayment);
+            }
+        }
+
+        allocateThePaymentAndMarkBulkScanPaymentAsProcessed(bulkScanPaymentRequestStrategic, paymentGroupReference, newPayment, headers);
+        return new ResponseEntity<>(paymentDtoMapper.toBulkScanPaymentStrategicDto(newPayment, paymentGroupReference), HttpStatus.CREATED);
+    }
+
+    @ApiOperation(value = "Record a Bulk Scan Payment with Payment Group", notes = "Record a Bulk Scan Payment with Payment Group")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Bulk Scan Payment with payment group created"),
+        @ApiResponse(code = 400, message = "Bulk Scan Payment creation failed"),
+        @ApiResponse(code = 422, message = "Invalid or missing attribute")
+    })
+    @PostMapping(value = "/payment-groups/bulk-scan-payments-strategic")
+    @ResponseBody
+    @Transactional
+    public ResponseEntity<PaymentDto> recordUnsolicitedBulkScanPaymentStrategic(@Valid @RequestBody BulkScanPaymentRequestStrategic bulkScanPaymentRequestStrategic
+        , @RequestHeader(required = false) MultiValueMap<String, String> headers) throws CheckDigitException {
```

Comment:
What is the purpose of propagating Http Headers received in request?

Reply:
with these changes bulk scan api called via payhub and not paybubble , propagation is related to that purpose

Reply:
see previous comment